### PR TITLE
Return backup stats in backup action

### PR DESCRIPTION
**Pull request by redgoat650**
## Change Overview

Backup action will also return backup size and file count. These data can be parsed from the backup execution output log. File count and backup size are returned as empty strings if parsing was unsuccessful.

## Pull request type

Please check the type of change your PR introduces:
- [ ] Work in Progress
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Trivial/Minor
- [ ] Bugfix
- [x] Feature
- [ ] Documentation

## Test Plan

Unit test to verify correct parsing of the backup output log
Adding checks to the outputs of the backup tests in the Data suite

- [ ] Manual
- [x] Unit test
- [ ] E2E

### CODE REVIEW

File: pkg/function/backup_data.go [hunk 1]
```diff
@@ -112,13 +112,15 @@ func (*backupDataFunc) Exec(ctx context.Context, tp param.TemplateParams, args m
...
 	if err != nil {
 		return nil, errors.Wrapf(err, "Failed to backup data")
 	}
 	output := map[string]interface{}{
-		BackupDataOutputBackupID:  backupID,
-		BackupDataOutputBackupTag: backupTag,
+		BackupDataOutputBackupID:       backupID,
+		BackupDataOutputBackupTag:      backupTag,
+		BackupDataStatsOutputFileCount: fileCount,
+		BackupDataStatsOutputSize:      backupSize,
```
Comment (redgoat650) [inline pkg/function/backup_data.go]: @tdmanv @DeepikaDixit - may need your opinion on this. I am using the same key constants defined for BackupStats for the two new fields. Is this appropriate, or should I define new constants for the BackupData version of FileCount and Size?

File: pkg/function/backup_data.go [hunk 2]
```diff
@@ -128,34 +130,36 @@ func (*backupDataFunc) RequiredArgs() []string {
...
 	}
 	// Get the snapshot ID from log
 	backupID := restic.SnapshotIDFromBackupLog(stdout)
 	if backupID == "" {
-		return "", "", errors.New("Failed to parse the backup ID from logs")
+		return "", "", "", "", errors.New("Failed to parse the backup ID from logs")
 	}
-	return backupID, backupTag, nil
+	// Get the snapshot ID from log
+	fileCount, backupSize := restic.SnapshotStatsFromBackupLog(stdout)
```
Comment (redgoat650) [inline pkg/function/backup_data.go]: Note I am not doing any error checking here. If the parse failed I'm returning the values as-are (empty strings).

File: pkg/function/backup_data.go (hunk 1 above)
Comment (tdmanv) [reply pkg/function/backup_data.go]: I think this is fine

File: pkg/restic/restic.go [hunk 3]
```diff
@@ -304,6 +304,23 @@ func SnapshotIDFromBackupLog(output string) string {
...
 
+// SnapshotStatsFromBackupLog gets the Snapshot file count and size from Backup Command log
+func SnapshotStatsFromBackupLog(output string) (fileCount string, backupSize string) {
+	if output == "" {
+		return "", ""
+	}
+	logs := regexp.MustCompile("[\n]").Split(output, -1)
+	for _, l := range logs {
+		// Log should contain "processed %d files, %.3f [Xi]B in mm:ss"
+		pattern := regexp.MustCompile(`processed\s([\d]+)\sfiles,\s([\d]+(\.[\d]+)?\s([TGMK]i)?B)\sin\s`)
```
[the remark this answers is not in the document]
Comment (redgoat650) [reply pkg/restic/restic.go]: good call, I'll move it for `SnapshotIDFromBackupLog` as well

File: pkg/restic/restic.go [hunk 4]
```diff
@@ -304,6 +304,23 @@ func SnapshotIDFromBackupLog(output string) string {
...
+		match := pattern.FindAllStringSubmatch(l, 1)
+		if match != nil {
+			return match[0][1], match[0][2]
```
[the remark this answers is not in the document]
Comment (redgoat650) [reply pkg/restic/restic.go]: Done, added bounds checks for `SnapshotIDFromBackupLog` too

File: pkg/function/backup_data.go [hunk 5]
```diff
@@ -128,34 +130,36 @@ func (*backupDataFunc) RequiredArgs() []string {
 		BackupDataIncludePathArg, BackupDataBackupArtifactPrefixArg}
 }
 
-func backupData(ctx context.Context, cli kubernetes.Interface, namespace, pod, container, backupArtifactPrefix, includePath, encryptionKey string, tp param.TemplateParams) (string, string, error) {
+func backupData(ctx context.Context, cli kubernetes.Interface, namespace, pod, container, backupArtifactPrefix, includePath, encryptionKey string, tp param.TemplateParams) (string, string, string, string, error) {
```
[the remark this answers is not in the document]
Comment (redgoat650) [reply pkg/function/backup_data.go]: yep I agree; will address this

File: pkg/function/backup_data.go (hunk 2 above)
Comment (tdmanv) [reply pkg/function/backup_data.go]: Let's at least log a failure to parse at debug level.

File: pkg/function/backup_data.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (redgoat650) [reply pkg/function/backup_data.go]: Should be addressed by this commit:
https://github.com/kanisterio/kanister/pull/337/commits/194afd72c0909784dfafde4c760f80d815570c2b

File: pkg/function/backup_data.go (hunk 2 above)
Comment (redgoat650) [reply pkg/function/backup_data.go]: log.Debug added

**Pull request merged**